core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -174,7 +171,8 @@ public void start(Listener<RespT> observer, Metadata headers) {
     }
 
     if (stream == null) {
-      stream = new DelayedStream(transportFuture, headers, listener);
+      stream = new DelayedStream(
+          this, callOptions, transportFuture, headers, listener, callExecutor, method);
```

Comment:
Could you point out which lock you are talking about? From my reading of the original code on line 476 there is a lock, and it is `this`, and `this` appears to be `DelayedStream`.